### PR TITLE
Add the ability to have Nginx version checks in templates

### PR DESCRIPTION
**Pull request by oseoin**
### Proposed changes

Add the ability to add version dependent template elements via a new `nginx.Version` struct which is available in the main

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by jjngx — APPROVED** [review]
👍🏻

File: internal/configs/version1/nginx-plus.tmpl [hunk 1]
```diff
@@ -346,3 +346,9 @@ stream {
 
     include /etc/nginx/stream-conf.d/*.conf;
 }
+
+{{- if and (.NginxVersion.IsPlus) (.NginxVersion.PlusGreaterThanOrEqualTo "nginx-plus-r31") }}
```
Comment (vepatel) [inline internal/configs/version1/nginx-plus.tmpl:350]: do we need `(.NginxVersion.IsPlus)` here as string in next condition is only possible on plus versions?

File: cmd/nginx-ingress/main.go [hunk 2]
```diff
@@ -79,7 +79,7 @@ func main() {
 		appProtectVersion = getAppProtectVersionInfo()
 	}
 
-	updateSelfWithVersionInfo(kubeClient, version, nginxVersion, appProtectVersion)
+	updateSelfWithVersionInfo(kubeClient, version, nginxVersion.String(), appProtectVersion)
```
Comment (vepatel) [inline cmd/nginx-ingress/main.go:82]: is its possible to check if pod labels are being assigned correctly after this change?

Comment (oseoin) [reply cmd/nginx-ingress/main.go:82]: Not sure if we have a test for it, but locally seems fine: `app.nginx.org/version: 1.25.3-nginx-plus-r31`

**Review by j1m-ryan — APPROVED** [review]

**Review by vepatel — APPROVED** [review]

**Pull request merged**